discovery-provider/src/queries/query_helpers.py

```diff
@@ -546,10 +547,24 @@ def get_repost_counts(session, query_by_user_flag, query_repost_type_flag, filte
         repost_counts_query = repost_counts_query.filter(
             Repost.blocknumber <= max_block_number
         )
+
+    return repost_counts_query
+
+def get_repost_counts(session, query_by_user_flag, query_repost_type_flag, filter_ids, repost_types, max_block_number=None):
+    repost_counts_query = get_repost_counts_query(session, query_by_user_flag, query_repost_type_flag, filter_ids, repost_types)
     return repost_counts_query.all()
 
 
-def get_save_counts(session, query_by_user_flag, query_save_type_flag, filter_ids, save_types, max_block_number=None):
+def get_repost_counts_with_time(session, query_by_user_flag, query_repost_type_flag, filter_ids, repost_types, time, max_block_number=None):
```

Comment:
yeah this make sense, but why add a separate function instead of a new argument to get_repost_counts? not saying we should refactor but jw on the functionality.
 
also - can we rename this from 'get_repost_counts_with_time' to 'get_repost_counts_for_time' since we aren't actually adding a new 'time' field to the repost counts, we are just filtering the results for a time period. correct me if i have misunderstood this functionality here

Reply:
it honestly feels cleaner to add a new parameter with a default value like max_block_number to the existing function and then enable this filter.

same for save_counts

Reply:
no you're right. there's no need to have another method when i can just add a default param of `time=None`. i forgot that you can use default params :) thanks for the catch! 

will make the change from using 'with' to 'for'

Reply:
looks good